### PR TITLE
use rapids-build-backend

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/31
Contributes to https://github.com/rapidsai/dependency-file-generator/issues/89

Proposes introducing `rapids-build-backend` as this project's build backend, to reduce the complexity of various CI/build scripts.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by jameslamb** [comment]
It looks like the `conda-python-tests` jobs are failing here, like this:

```text
FAILED tests/basicpreds/test_intersections.py::test_one_pair - AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'pandas.core.indexes.base.Index'> instead
FAILED tests/basicpreds/test_intersections.py::test_two_pairs - AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'pandas.core.indexes.base.Index'> instead
FAILED tests/basicpreds/test_intersections.py::test_one_pair_with_overlap - AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'pandas.core.indexes.base.Index'> instead
FAILED tests/basicpreds/test_intersections.py::test_two_pairs_with_intersect_and_overlap - AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'pandas.core.indexes.base.Index'> instead
FAILED tests/basicpreds/test_intersections.py::test_one_pair_multilinestring - AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'pandas.core.indexes.base.Index'> instead
FAILED tests/basicpreds/test_intersections.py::test_three_pairs_identical_has_ring - AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'pandas.core.indexes.base.Index'> instead
FAILED tests/basicpreds/test_intersections.py::test_three_pairs_identical_no_ring - AssertionError: Series Expected type <class 'pandas.core.series.Series'>, found <class 'pandas.core.indexes.base.Index'> instead
```

([build link](https://github.com/rapidsai/cuspatial/actions/runs/9392305018/job/25866740018?pr=1393))

I see the same failures on 1384 ([build link](https://github.com/rapidsai/cuspatial/actions/runs/9402555342/job/25897932693?pr=1384)), so fairly confident that they're not related to the changes here.

File: dependencies.yaml [hunk 1]
```diff
@@ -361,7 +383,7 @@ dependencies:
     common:
       - output_types: conda
         packages:
-          - &rmm_conda rmm==24.8.*
+          - &rmm_conda rmm==24.8.*,>=0.0.0a0
```
Comment (isVoid) [inline dependencies.yaml:386]: What does this `0.0.0a0` mean?

Comment (jameslamb) [reply dependencies.yaml:386]: **short answer**

For conda, it's identical to just `==24.8.*`.

For `pip` it means "I'm willing to accept any pre-releases of this package as well".

**long answer**

A version number like that is called a "pre-release".

Those are the version numbers we give to nightly RAPIDS packages.

By default, when you run `pip install cuml-cu12==24.8.*`, such versions will be ignored. That's so package authors can publish packages like release candidates for testing.

One option to tell `pip` you're open to installing such packages is to pass the `--pre` flag to `pip install`, like this:

```shell
pip install --pre 'cuml-cu12==24.8.*`
```

But that can have unintended consequences... `--pre` applies through the entire dependency tree, so running a command like that would also mean you're saying "I'm willing to pull in any pre-releases of any of `cuml-cu12`'s dependencies as well".

Specifying a constraint like `,>=0.0.0a0` like this is a bit stricter... it tells `pip` "I'm willing to pull in any pre-releases of `cuml-cu12` specifically, but not necessarily its dependencies".

There's some more discussion on this in https://peps.python.org/pep-0440/#handling-of-pre-releases

**Review by bdice — APPROVED** [review]
Thanks!

File: dependencies.yaml (hunk 1 above)
Comment (vyasr) [reply dependencies.yaml:386]: We should probably update https://github.com/rapidsai/build-planning/blob/main/docs/docs/packaging.md#nightlies (and then use it as a reference).

**Review by vyasr — APPROVED** [review]

**Pull request merged**